### PR TITLE
fix(x/ecocredit): charge single basket fee and migrate param

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1165 / Closes: #1166 / Ref: #1183

This pull request updates the fee requirements for basket creation to accept only one fee in `MsgCreate`. We are not bumping the basket package for v4.0.0 so instead we require no more than one fee in basic message validation and we use the minimum fee when checking the balance, sending to the module, and burning.

This pull request also sets the basket fee in the upgrade handler (again) because the field name has been updated but the key was still `BasketCreationFee`. To prevent confusion, the key is now `BasketFee` and matches the field name and in order to do so, it looks like we have to reset the basket fee in the upgrade handler but maybe there is another way.

Notes from manual testing: https://hackmd.io/wqnic3hIQtqndGyXi62zgQ

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
Notes from manual testing: https://hackmd.io/wqnic3hIQtqndGyXi62zgQ

File: x/ecocredit/server/basket/msg_create.go [hunk 1]
```diff
@@ -19,31 +19,56 @@ func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgC
...
-	if err != nil {
-		return nil, err
+		// find the minimum fee
+		for _, coin := range fee {
+			if coin.Denom == msg.Fee[0].Denom {
+				if minimumFee.IsNil() || coin.IsLT(minimumFee) {
+					minimumFee = coin
+				}
+			}
+		}
```
Comment (technicallyty) [inline x/ecocredit/server/basket/msg_create.go:54]: the `validateBasketFee` function calls `coins.Validate` which will error on duplicate denoms. `fee.AmountOf(msg.Fee[0].Denom)` should suffice here

File: x/ecocredit/server/basket/msg_create.go [hunk 2]
```diff
@@ -19,31 +19,56 @@ func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgC
...
+
+		curatorBalance := k.bankKeeper.GetBalance(sdkCtx, curator, minimumFee.Denom)
+		if curatorBalance.IsNil() || curatorBalance.IsLT(minimumFee) {
+			return nil, sdkerrors.ErrInsufficientFunds.Wrapf("insufficient balance for bank denom %s", minimumFee.Denom)
+		}
+
+		minimumFees := sdk.NewCoins(minimumFee)
```
Comment (technicallyty) [inline x/ecocredit/server/basket/msg_create.go:61]: might be better to just use `sdk.Coins{minimumFee}`  here, `NewCoins` runs all the validation again

Comment (ryanchristo) [reply x/ecocredit/server/basket/msg_create.go]: https://github.com/regen-network/regen-ledger/pull/1184/commits/44db6295c6bfd2bcb3d86e49260c7b599e312549

File: x/ecocredit/server/basket/msg_create.go (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/server/basket/msg_create.go]: https://github.com/regen-network/regen-ledger/pull/1184/commits/44db6295c6bfd2bcb3d86e49260c7b599e312549

**Review by technicallyty — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**